File: novachat-endstone/novachat_endstone/extension/extension_exception.py
"""
Extension exception module.

This module defines the exception class for extension-related errors.

Requirements: 10.2 - THE Endstone 扩展 SHALL 使用 Python 编写并放置在 extensions 目录
"""

from typing import Optional


class ExtensionException(Exception):
    """
    Exception thrown when an extension fails to load or encounters an error.
    """
    
    def __init__(
        self, 
        message: str, 
        extension_id: Optional[str] = None,
        cause: Optional[Exception] = None
    ):
        """
        Creates a new ExtensionException.
        
        Args:
            message: The error message
            extension_id: The ID of the extension that caused the error
            cause: The underlying exception that caused this error
        """
        self.extension_id = extension_id
        self.cause = cause
        
        full_message = f"Extension '{extension_id}': {message}" if extension_id else message
        super().__init__(full_message)

File: novachat-pmmp/src/NovaChat/Extension/ExtensionException.php
<?php

declare(strict_types=1);

namespace NovaChat\Extension;

use Exception;

/**
 * Exception thrown when an extension fails to load or encounters an error.
 * 
 * Requirements: 10.1 - THE PMMP 扩展 SHALL 使用 PHP 编写并放置在 extensions 目录
 */
class ExtensionException extends Exception {
    
    private ?string $extensionId;
    
    /**
     * Creates a new ExtensionException.
     * 
     * @param string $message the error message
     * @param string|null $extensionId the ID of the extension that caused the error
     * @param \Throwable|null $previous the previous exception
     */
    public function __construct(string $message, ?string $extensionId = null, ?\Throwable $previous = null) {
        $this->extensionId = $extensionId;
        
        $fullMessage = $extensionId !== null 
            ? "Extension '$extensionId': $message" 
            : $message;
        
        parent::__construct($fullMessage, 0, $previous);
    }
    
    /**
     * Gets the ID of the extension that caused the error.
     * 
     * @return string|null the extension ID, or null if not applicable
     */
    public function getExtensionId(): ?string {
        return $this->extensionId;
    }
}
